src/hffs/fs.py

```diff
@@ -155,33 +167,34 @@ def _open(
         self,
         path: str,
         mode: str = "rb",
+        revision: Optional[str] = None,
         **kwargs,
     ):
         if mode == "ab":
             raise NotImplementedError("Appending to remote files is not yet supported.")
         path = self._strip_protocol(path)
-        return HfFile(self, path, mode=mode, **kwargs)
+        return HfFile(self, path, mode=mode, revision=revision, **kwargs)
 
-    def _rm(self, path, **kwargs):
-        repo_type, repo_id, path_in_repo = self._resolve_repo_id(path)
+    def _rm(self, path, revision: Optional[str] = None, **kwargs):
         path = self._strip_protocol(path)
+        repo_type, repo_id, revision, path_in_repo = self._resolve_path_with_revision(path, revision)
         operations = [huggingface_hub.CommitOperationDelete(path_in_repo=path_in_repo)]
         commit_message = f"Delete {path}"
         self._api.create_commit(
             repo_id=repo_id,
             repo_type=repo_type,
             token=self.token,
             operations=operations,
-            revision=self.revision,
+            revision=revision,
             commit_message=kwargs.get("commit_message", commit_message),
             commit_description=kwargs.get("commit_description"),
         )
-        self.invalidate_cache(path=path)
+        self.invalidate_cache(path=self._unresolve_path(repo_type, repo_id, revision, path_in_repo))
```

Comment:
Typically with a dataclass you could do something like `self.invalidate_cache(path=resolved_path.unresolve()))`. But still feel free to ignore if you prefer as it is now